### PR TITLE
feat(TFD-5486): Rewrite Window processor

### PR DESCRIPTION
**Pull request by amaumont**
migrate Window processor from TCOMPV0 to Tacokit with unit tests

### CODE REVIEW

File: processing-beam/src/main/java/org/talend/components/processing/window/WindowConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    @Max(Integer.MAX_VALUE)
+    @Documentation("The window duration (Data during X ms)")
+    private Integer windowLength = 5000;
+
+    @Option
+    @Required
+    @Min(0)
+    @Max(Integer.MAX_VALUE)
+    @Documentation("The window slide length (Every X ms)")
+    private Integer windowSlideLength = 5000;
```
Comment (bchen-talend) [inline processing-beam/src/main/java/org/talend/components/processing/window/WindowConfiguration.java]: need to hidden when session checked
```suggestion
    @ActiveIf(target = "windowSession", value = "false")
    private Integer windowSlideLength = 5000;
```

File: processing-beam/src/main/resources/org/talend/components/processing/window/Messages.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+Processing.Window._displayName = Window
```
Comment (bchen-talend) [inline processing-beam/src/main/resources/org/talend/components/processing/window/Messages.properties]: better to specify _displayName for each property, else UI will show the var name.
WindowConfiguration.windowLength._displayName
WindowConfiguration.windowSlideLength._displayName
WindowConfiguration.windowSession._displayName

**Comment by bchen-talend** [comment]
I didn't check jira description before the above comment, so you may waiting for answer about the new `Window type` and other property name.

File: processing-beam/src/main/java/org/talend/components/processing/window/WindowConfiguration.java (hunk 1 above)
Comment (amaumont) [reply processing-beam/src/main/java/org/talend/components/processing/window/WindowConfiguration.java]: Good point.

**Comment by amaumont** [comment]
Required to create pull request from 'dss/tacokit_integration' not from 'master'

**Pull request closed without merging**